File: app/src/main/java/com/mintthursday/StepsShowAdapter.java
package com.mintthursday;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.LinearLayoutCompat;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class StepsShowAdapter extends RecyclerView.Adapter<StepsShowAdapter.StepsShowViewHolder>{
    private List<Step> stepsList = new ArrayList<>();
    @NonNull
    @Override
    public StepsShowAdapter.StepsShowViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.view_step_show, parent, false);
        return new StepsShowViewHolder(view);
    }
    public void setItems(List<Step> steps) {
        stepsList = steps;
        notifyDataSetChanged();
    }

    @Override
    public void onBindViewHolder(@NonNull StepsShowAdapter.StepsShowViewHolder holder, int position) {
        holder.bind(stepsList.get(position));
    }

    @Override
    public int getItemCount() {
        return stepsList.size();
    }

    public class StepsShowViewHolder extends RecyclerView.ViewHolder{
        TextView textView;

        public StepsShowViewHolder(@NonNull View itemView) {
            super(itemView);
            textView = itemView.findViewById(R.id.step_show);
        }
        public void bind(Step step) {
            textView.setText(step.getStepInstruction());
        }
    }
}

File: app/src/main/java/com/mintthursday/recipe/show/steps/StepsShowAdapter.kt
package com.mintthursday.recipe.show.steps

import androidx.recyclerview.widget.RecyclerView
import com.mintthursday.recipe.show.steps.StepsShowAdapter.StepsShowViewHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.mintthursday.R
import android.widget.TextView
import com.mintthursday.models.Step
import java.util.ArrayList

class StepsShowAdapter : RecyclerView.Adapter<StepsShowViewHolder>() {
    private var items: List<Step> = ArrayList()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StepsShowViewHolder {
        return StepsShowViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.view_step_show, parent, false))
    }

    fun setItems(steps: List<Step>) {
        items = steps
        notifyItemRangeInserted(0, items.size)
    }

    override fun onBindViewHolder(holder: StepsShowViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int {
        return items.size
    }

    class StepsShowViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var textView: TextView = itemView.findViewById(R.id.step_show)
        fun bind(step: Step) {
            textView.text = step.stepInstruction
        }

    }
}